src/js/BookReader.js

```diff
@@ -277,7 +283,11 @@ BookReader.prototype.initParams = function() {
     // params explicitly set in URL take precedence over all other methods
     var urlParams = this.paramsFromFragment(this.urlReadFragment());
     if (urlParams.mode) {
-      this.prevReadMode = urlParams.mode;
+      // If there were any parameters
+      if (Object.keys(urlParams).length > 0) {
+        this.extendParams(params, urlParams);
```

Comment:
Why is there a second call to `extendParams` here? Just outside of this condition, the same call is made. It looks like you could set `params.fragmentChange = !!Object.keys(urlParams).length` and drop the condition.

Reply:
fixed